planner/types.ts

```diff
@@ -0,0 +1,161 @@
+export class Activity {
+    name: string;
+    duration: number;
+    NoP: number;
+    role: Role;
+    input: IOSet[];
+    output: IOSet[];
+
+    public constructor(name: string, duration: number, NoP: number, role: Role, input: IOSet[], output: IOSet[]) {
+        this.name = name;
+        this.duration = duration;
+        this.NoP = NoP;
+        this.role = role;
+        this.input = input;
+        this.output = output;
+    }
+
+    public isExecutable(currentState: State[], ressources: Resource[]): boolean {
+        return this.input.some(inputSet => inputSet.satisfiedBy(currentState)) && ressources.some(resource => resource.satisfies(this.role, this.NoP));
+    }
+}
+
+export class IOSet {
+    set: Array<{ dataObject: State, isList: boolean }>;
+
+    public constructor(set: Array<{ dataObject: State, isList: boolean }>) {
+        this.set = set;
+    }
+
+    public satisfiedBy(currentState: State[]): boolean {
+        let satisfiedBy = true;
+        this.set.forEach(function (State) {
+                if (!currentState.includes(State.dataObject)) {
+                    satisfiedBy = false;
+                }
+            }
+        );
+        return satisfiedBy;
+    }
+}
+
+export class Role {
+    name: string;
+
+    public constructor(name: string) {
+        this.name = name;
+    }
+}
+
+export class Resource {
+    name: string;
+    role: Role;
+    capacity: number;
+
+    public constructor(name: string, role: Role, capacity: number) {
+        this.name = name;
+        this.role = role;
+        this.capacity = capacity;
+    }
+
+    public satisfies(role: Role, NoP: number): boolean {
+        return role === this.role && NoP <= this.capacity;
+    }
+}
+
+export class Dataclass {
+    name: string;
+
+    public constructor(name: string) {
+        this.name = name;
+    }
+}
+
+export class State {
```

Comment:
You could call it `DataObjectNode` with the attributes `class`, `state`, and `isList` to make sure it maps to the nodes in the fragments.